### DISCUSSION TITLE
Auto-reloading the page after new SSG / generate

### DISCUSSION THREAD

**Discussion by fabianwohlfart** [nuxt/nuxt#29665] [Questions] [answered]
How does Nuxt handle it when a user is browsing the site and a new SSG build gets uploaded? Do they end up hitting a bunch of 404s, or is there some kind of automatic page reload when the build hashes don’t match anymore?

**Comment by TheAlexLichter** [comment]
See https://github.com/nuxt/nuxt/issues/14594 + linked issues ☺️
There is a mechanism for it 👍

**Reply by fabianwohlfart** [reply]
Thanks, but this issue is so confusing with all the questions and answers and no marked solution. Does it mean there is an in-build solution for this, or do I have to implement it myself? If so can I find it in the docs somewhere? – I was searching for it but with no luck.

**Reply by fabianwohlfart** [reply]
Here is what I get:

1. npm run generate
2. serve and visit site
3. change something in a component
4. npm run generate
5. navigate on the site

![Bildschirmfoto 2024-10-24 um 12 22 02](https://github.com/user-attachments/assets/272e8231-cdf3-455a-9b46-6de6f7a99b03)
![Bildschirmfoto 2024-10-24 um 12 22 24](https://github.com/user-attachments/assets/4ab359c9-0675-499a-8b44-30303a2bf227)

So i guess it's not in-build and i have to write it myself?

**Reply by fabianwohlfart** [reply, marked as answer]
Did some more research with the keywords you gave me and I think it is this PR https://github.com/nuxt/nuxt/pull/28160, am I correct?

**Reply by TheAlexLichter** [reply]
Yes!
